subsys/tracing/user/tracing_user.c

```diff
@@ -25,6 +25,18 @@ void __weak sys_trace_isr_exit_user(void) {}
 void __weak sys_trace_idle_user(void) {}
 void __weak sys_trace_sys_init_enter_user(const struct init_entry *entry, int level) {}
 void __weak sys_trace_sys_init_exit_user(const struct init_entry *entry, int level, int result) {}
+void __weak sys_trace_gpio_pin_active_user(const struct device *port, gpio_pin_t pin) {}
+void __weak sys_trace_gpio_pin_inactive_user(const struct device *port, gpio_pin_t pin) {}
+void __weak sys_trace_gpio_pin_configured_output_user(const struct device *port, gpio_pin_t pin,
```

Comment:
@nashif any idea about these compliance errors? do we have to follow the recommendation but break consistency with the existing declarations?